File: pipeline-test/src/main/java/helpers/UrlBuilder.java
package helpers;
import com.pipe.pipeapp.Configuration;
import com.pipe.pipeapp.SessionStream;
import org.testng.annotations.Optional;

import java.text.ParseException;

public class UrlBuilder {

    StringBuilder authUrl = new StringBuilder();
    public StringBuilder buildAuthUrl(String baseAuthUrl,int pid, int subsId,String msgSize) {

        if (baseAuthUrl.equals("")) {
            authUrl.append(defaultSettingAuthUrlName());

        } else {
            authUrl.append(baseAuthUrl);

        }

        if(pid==-1) {
            authUrl.append("?pid=" + defaultSettingPID() + "");
        }else{
            authUrl.append("?pid="+pid);
        }

        authUrl.append("&as=1");

     //   authUrl.append("&skiprnd=1");

        if(subsId == -1){
            authUrl.append("&subsid=" + defaultSettiongsubsId());

        }else{
            authUrl.append("&subsid=" + (Integer.toString(subsId)+""));
        }

        authUrl.append("&msgsize="+defaultSettingMsgSize());

        System.out.println(authUrl);
        return authUrl;
    }

    private String defaultSettiongsubsId(){
        return Configuration.prop.getProperty("suite.pusher.defaultParams.subsId");
    }
    private String defaultSettingMsgSize(){
        return Configuration.prop.getProperty("suite.pusher.defaultParams.msgSize");
    }

    private String defaultSettingPID() {
        return Configuration.prop.getProperty("suite.pusher.defaultParams.optionalPid");
    }

    private String defaultSettingAuthUrlName () {
        return Configuration.prop.getProperty("suite.pusher.defaultParams.baseAuth");
    }

    private String defaultSettingDnsName(){
        return Configuration.prop.getProperty("suite.pusher.defaultParams.dnsName");
    }

    public StringBuilder buildStreamUrl(StringBuilder dnsName){
        return dnsName.append("wr/?");
    }

//    public String buildStreamV16(String sid, SessionStream streamData, int subsId, String dnsName) {
//        StringBuilder finalDnsName = new StringBuilder();
//
//        if (dnsName == "") {
//            finalDnsName.append(defaultSettingDnsName());
//        } else {
//            finalDnsName.append(dnsName);
//        }
//        String stream = buildStreamUrl(finalDnsName) + sid + "&" + defaultSettingPID() + "&" + "10" + "&" + Integer.toString(streamData.messageNumber) + "&" + Integer.toString(streamData.streamId) + "&" +
//                Integer.toString(streamData.streamMessageId) + "&" + Integer.toString(streamData.dataFlagType + 256);
//
//        if (subsId == -1) {
//        }
//
//    }

     public String buildStream(String sid, SessionStream streamData, int pid, int subsId, String dnsName){


        if(dnsName.equals("")){
            dnsName=defaultSettingDnsName();
        }

        if(pid == -1){
           pid=Integer.parseInt(defaultSettingPID());
        }

        if(subsId == -1)
            subsId=Integer.parseInt(defaultSettiongsubsId());

        String streamUrl = dnsName + "wr/?" + sid + "&" + Integer.toString(pid) + "&f&" + Integer.toString(streamData.getMessageNumber()) +
                "&" + Integer.toString(streamData.getStreamId()) + "&" + Integer.toString(streamData.getStreamMessageId()) + "&" + Integer.toString(streamData.getDataFlagType()) +
                "&subsid=" + Integer.toString(subsId);

        return streamUrl;

    }
}

File: Regressions/src/main/scala/com/clicktale/pipeline/framework/helpers/UrlBuilder.scala
package com.clicktale.pipeline.framework.helpers

import java.io.{File, PrintWriter}

import com.clicktale.pipeline.dataObjects.SessionStream
import com.clicktale.pipeline.framework.dal.ConfigParser.conf


object UrlBuilder {


  def retry[T](times: Int)(fn: => T): Option[T] =
    (1 to times).view flatMap (n => try Some(fn) catch {
      case e: Exception => None
    }) headOption

  def defaultSettingsPID(): Int =
    conf.getString(s"WebRecorder.Environments.${conf.getString("WebRecorder.Current.Environment")}.Pid").toInt

  def defaultSettingsProtocol(): String =
    conf.getString("WebRecorder.Protocol")

  def defaultSettingsDnsName(): String =
    conf.getString(s"WebRecorder.Environments.${conf.getString("WebRecorder.Current.Environment")}.BaseUrl")

  def buildAuthUrl(optionalMobile: Boolean = false,
                   pid: Int = -1,
                   subsId: Int = -1,
                   uid: Long = -1,
                   authSuffix: String = "",
                   dnsName: String = ""): String = {

    val authUrl =
      (if (dnsName == "") defaultSettingsDnsName() else dnsName) +
      (if (pid == -1) "auth?pid=" + defaultSettingsPID() else "auth?pid=" + pid) +
      (if(subsId == -1) "" else s"&subsid=${subsId.toString}") +
      (if (uid == (-1)) "" else s"&uid=${uid.toString}") +
      (if (optionalMobile) s"&m=1" else "") +
      authSuffix

    println(authUrl)
    authUrl
  }

  def buildStreamUrl(dnsName: String): String = {
    dnsName + "wr/?"
  }

  def buildStream(sid: String,
                  streamData: SessionStream,
                  optionalPid: Int = -1,
                  subsId : Int = -1,
                  dnsName: String = "",
                  isV16Protocol: Boolean = false,
                  optionalUrlSuffix:String = ""): String = {

    val finalDnsName: String =
      if (dnsName == ""){defaultSettingsDnsName()}
      else dnsName

    val streamDataFlag: String =
      if (isV16Protocol) (streamData.dataFlagType + 256).toString
      else streamData.dataFlagType.toString

    val pId = if (optionalPid == -1) defaultSettingsPID() else optionalPid
    val streamUrl = buildStreamUrl(finalDnsName) + sid + "&" + pId.toString + "&" + defaultSettingsProtocol() + "&" +
      streamData.messageNumber.toString + "&" + streamData.streamId.toString + "&" +
      streamData.streamMessageId.toString + "&" + streamDataFlag +
      (if(subsId==(-1)) "" else "&subsid="+subsId.toString) +
      optionalUrlSuffix
    streamUrl
  }

  def buildStreamV16(sid: String,
                     streamData: SessionStream,
                     subsId: Int = -1,
                     dnsName: String = ""): String = {

    val finalDnsName: String =
      if (dnsName == ""){defaultSettingsDnsName()}
      else dnsName

    val stream =
      buildStreamUrl(finalDnsName) + sid + "&" + defaultSettingsPID().toString + "&" + "10" + "&" +
        streamData.messageNumber.toString + "&" + streamData.streamId.toString + "&" +
        streamData.streamMessageId.toString + "&" + (streamData.dataFlagType + 256).toString +
        (if(subsId==(-1)) "" else "&subsid="+subsId.toString)
    stream
  }


  def writeToFile(filename: String, content: String): Unit = {
    val writer = new PrintWriter(new File(filename))
    writer.write(content)
    writer.close()
  }
}
